pkg/cmd/server/origin/master.go

```diff
@@ -411,35 +411,60 @@ func (c *MasterConfig) ensureMasterAuthorizationNamespace() {
 // ensureComponentAuthorizationRules initializes the global policies
 func (c *MasterConfig) ensureComponentAuthorizationRules() {
 	registry := authorizationetcd.New(c.EtcdHelper)
-	ctx := kapi.WithNamespace(kapi.NewContext(), c.MasterAuthorizationNamespace)
 
-	if existing, err := registry.GetPolicy(ctx, authorizationapi.PolicyName); err == nil || strings.Contains(err.Error(), " not found") {
-		if existing != nil && existing.Name == authorizationapi.PolicyName {
-			return
-		}
+	roleRegistry := roleregistry.NewVirtualRegistry(registry)
+	for _, role := range bootstrappolicy.GetBootstrapRoles(c.MasterAuthorizationNamespace, c.OpenshiftSharedResourcesNamespace) {
+		ctx := kapi.WithNamespace(kapi.NewContext(), role.Namespace)
 
-		bootstrapGlobalPolicy := authorizer.GetBootstrapPolicy(c.MasterAuthorizationNamespace)
-		if err = registry.CreatePolicy(ctx, bootstrapGlobalPolicy); err != nil {
-			glog.Errorf("Error creating policy: %v due to %v\n", bootstrapGlobalPolicy, err)
-		}
+		if _, err := roleRegistry.GetRole(ctx, role.Name); kapierror.IsNotFound(err) {
+			if err := roleRegistry.CreateRole(ctx, &role); err != nil {
```

Comment:
Correct.  We're only writing roles when those roles are missing.  I can envision an improvement later on to sync to exact value for a role, but I want to be sure that we have a way to avoid doing that by default.